vertical-pod-autoscaler/updater/README.md

```diff
@@ -0,0 +1,31 @@
+# Vertical Pod Autoscaler - Updater
+
+# Introduction
+Updater component for Vertical Pod Autoscaler described in https://github.com/kubernetes/community/pull/338
+
+Updater runs in Kubernetes cluster and decides which pods should be restarted 
+based on resources allocation recommendation calcualted by Recommender.
+If a pod should be updated, Updater will try to evict the pod. 
+It respects the pod disruption budget, by using Eviction API to evict pods
+Updater does not perform the actual resources update, but relies on Vertical Pod Autoscaler admission plugin 
+to update pod resources when the pod is recreated after eviction.
+
+
+# Current implementation
+Runs in a loop. On one iteration performs:
+* Fetching Vertical Pod Autoscaler configuration - using mocked Lister implementation
+* Fetching live pods information with current resource allocation.
+* For each replicated pod spec fetching resources alocation recommendation - using mock api.
+* Recommendations are cached with ttl (specified by a flag)
+* For each replicated pods group calculating if pod update is required and how many replicas can be evicted. 
+Maximum ratio of evicted replicas is specified by flag.
+* Evicting pods if recommended resources significantly vary from the actual resources allocation.
+Threshold for evicting pods is specified by flag as percentage of resource that changed (i.e changes smaller than 10% are ignored)
+Priority of evictions within a set of replicated pods is proportional to sum of percentages of changes in resources 
+(i.e. pod with 15% memory increase 15% cpu decrease recommended will be evicted
+before pod with 20% memory increase and no change in cpu)
```

Comment:
We compare evictions priority only within a replica set. I think it will be a rare case that some pods should be shrinked and some expanded within the same replica set. This would mean that we have a new recommendation before previous recommendation was applied to all pods in replica set.  If that happens, expanding may be more important than shrinking. We can easily add a flag here controlling priority of expand vs shrink, but for now I would leave it in simple initial version.

This may be a discussion related more to a recommender module, but maybe we should somehow avoid giving new recommendation when pods are updating ?